File: Java/Medium/2109.java
class Solution {
    public String addSpaces(String s, int[] spaces) {
        StringBuilder answer = new StringBuilder();

        // index of the spaces
        int k = 0;
        for (int i = 0; i < s.length(); i++) {
            if (k < spaces.length && i == spaces[k]) {
                answer.append(" ");
                k++;
            }
            answer.append(s.charAt(i));
        }

        return answer.toString();
    }
}

File: Python/Medium/2109.py
class Solution:
    def addSpaces(self, s: str, spaces: List[int]) -> str:
        answer = []

        # for keeping track of answer index
        j=0

        # for keeping track of spaces index
        k=0
        for i in range(len(s)):
            if(k<len(spaces) and i==spaces[k]):
                answer.append(' ')
                k+=1

            answer.append(s[i])
            

        return "".join(answer)
